PTP-only documentation improvements (#1188)
suse-edge/suse-edge.github.io: asciidoc/product/atip-automated-provision.adoc

$ kubectl apply -f capi-provisioning-example.yaml
----

[#add-telco-ptp]
=== Precision Time Protocol (PTP)

PTP synchronization is enabled at provisioning time in two steps: enabling the daemons
in the EIB image and deploying the configuration files through Cluster API. Two
provisioning models are documented, depending on the target platform:

* <<add-telco-ptp-standard,Standard scenarios>>: a single `ptp4l` instance and, optionally,
`phc2sys`. Covers <<ptp-scenario-oc,Scenario 1>> through
<<ptp-scenario-bc-single-nic,Scenario 3>>.
* <<add-telco-ptp-gnrd,GNR-D multi-NIC scenarios>>: several `ptp4l` instances, `ts2phc` and
on-board DPLL setup, which require additional files and services. Covers
<<ptp-scenario-bc-multi-nic-dpll,Scenario 4>> (and the SyncE-enhanced Scenario 5).

[#add-telco-ptp-standard]
==== Standard scenarios (single ptp4l + phc2sys)

Only the configuration files change between these PTP scenarios, so a single generic

> example is used here and each scenario provides its specific values.

Reviewer: ```suggestion
Since only the configuration files change between these PTP scenarios, this section provides a single generic template followed by the specific values for each scenario.
```

Author: Done — reworded as suggested.

                  OPTIONS="${PHC2SYS_OPTIONS}"
              mode: 0644
              user:
                name: root
              group:
                name: root
    kubelet:
      extraArgs:
        - provider-id=metal3://BAREMETALHOST_UUID
    nodeName: "localhost.localdomain"
----

The placeholders are provided by each scenario:

[cols="1,2"]
|===
| Placeholder | Description

| `$\{PTP4L_CONF_PATH\}`
| Path of the `ptp4l` profile file, for example `/etc/ptp4l-G.8275.1.conf`.

| `$\{PTP4L_CONF_CONTENTS\}`
| Contents of that `ptp4l` file, taken from the chosen scenario.

| `$\{PTP4L_OPTIONS\}`
| Value of the `OPTIONS=` line in `/etc/sysconfig/ptp4l`.

| `$\{PHC2SYS_OPTIONS\}`
| Value of the `OPTIONS=` line in `/etc/sysconfig/phc2sys`.
|===

For the specific values to substitute in each case, see the *CAPI automation* of the
corresponding scenario in <<ptp-scenarios,the PTP scenarios>> of the PTP documentation.

[NOTE]
====
This approach is convenient only if the hardware in the cluster is uniform and the
same PTP configuration is needed on all hosts, interface name included. The Boundary
Clock scenarios that rely on `ts2phc` or on-board DPLL routing require additional

> files and services on top of this base example, described in the next section.

Reviewer: ```suggestion
files and services in addition to the base example, which are described in the next section.
```

Author: Done.

====

[#add-telco-ptp-gnrd]
==== GNR-D multi-NIC scenarios (DPLL)

The GNR-D Boundary Clock (<<ptp-scenario-bc-multi-nic-dpll,Scenario 4>>) cannot be
described with a single `ptp4l` profile: it runs one `ptp4l` instance per PHC, aligns the
add-in cards with `ts2phc`, wires the on-board DPLL at boot and forwards the Grand Master
parameters periodically. The provisioning therefore extends the base example with several
configuration files, scripts and `systemd` units, all deployed through the same

> `RKE2ControlPlane` Ignition block. The bodies of every file and unit are the ones listed in

Reviewer: Here, bodies refer to the contents? 
```suggestion
`RKE2ControlPlane` Ignition block. The contents of every file and unit are the ones listed in
```

Author: Yes, it refers to the contents. Changed `bodies` → `contents`.

<<ptp-scenario-bc-multi-nic-dpll,Scenario 4>>.

*Enable the daemons in the EIB image*

The GNR-D units are custom (a DPLL setup service, one templated `ptp4l` instance per PHC and
a Grand Master forwarding timer), so they are both deployed and enabled through the Ignition

> block below rather than through the EIB `systemd.enable` list. The EIB image only needs the

Reviewer: ```suggestion
The GNR-D units require a custom setup: a DPLL service, one templated `ptp4l` instance per PHC, and a Grandmaster forwarding timer. Consequently, you must deploy and enable them via the Ignition block below instead of using the EIB `systemd.enable` list.The EIB image only needs the
```

Author: Done. Applied your rewording (added the missing space after `list.`). Note: `Grandmaster` is now a single word throughout the doc — see the terminology note on the master/receiver thread.

`linuxptp` package, which provides the `ptp4l`, `ts2phc` and `pmc` binaries:

[,yaml,subs="attributes"]
----
operatingSystem:
  packages:
    packageList:
      - linuxptp      # already present in the Telco package list
    sccRegistrationCode: $SCC_REGISTRATION_CODE
----

*Deploy the files and units through Cluster API*

The `RKE2ControlPlane` object below deploys the GNR-D files and units. It follows the same

> approach as the standard example: the file bodies are provided as placeholders and taken from

Reviewer: For my understanding, what are file bodies? Can we replace that with file contents?

Author: Right — changed `file bodies` → `file contents` here too, for consistency.
